annlite/executor.py

```diff
@@ -260,22 +246,38 @@ def search(
             return
 
         limit = int(parameters.get('limit', self.limit))
-        search_filter = parameters.get('filter', {})
-        include_metadata = bool(
-            parameters.get('include_metadata', self.include_metadata)
-        )
-
+        search_filter = parameters.get('filter', None)
         access_paths = parameters.get('access_paths', self.search_access_paths)
         flat_docs = docs[access_paths]
-        if len(flat_docs) == 0:
-            return
+
         with self._index_lock:
-            self._index.search(
-                flat_docs,
-                filter=search_filter,
-                limit=limit,
-                include_metadata=include_metadata,
-            )
+            if len(self._data_buffer) > 0:
+                raise RuntimeError(
+                    f'Cannot search documents while the pending documents in the buffer are not indexed yet. '
+                    'Please wait for the pending documents to be indexed.'
+                )
```

Comment:
can we check that this does not stop the Executor ?

Reply:
The runtime error will not stop the executor. Here, we want to raise this error to notify the client (it will receive the exception message) what happened.